dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/MetadataValueRestRepository.java

```diff
@@ -183,92 +157,21 @@ public Page<MetadataValueWrapperRest> findByValue(@Parameter(value = "schema", r
-        }
-
-        return numberList;
-    }
-
     public List<MetadataValueWrapper> filterEUSponsors(List<MetadataValueWrapper> metadataWrappers) {
         return metadataWrappers.stream().filter(m -> !m.getMetadataValue().getValue().contains("info:eu-repo"))
                 .collect(Collectors.toList());
     }
 
     public List<MetadataValueWrapper> distinctMetadataValues(List<MetadataValueWrapper> metadataWrappers) {
         return metadataWrappers.stream().filter(
-                distinctByKey(metadataValueWrapper -> metadataValueWrapper.getMetadataValue().getValue()) )
+                Utils.distinctByKey(metadataValueWrapper -> metadataValueWrapper.getMetadataValue().getValue()) )
                 .collect( Collectors.toList() );
     }
 
-    /**
-     * From list of String and list of Numbers create a query for the SolrQuery.
-     * @param metadataField e.g. `dc.contributor.author`
-     * @param characterList e.g. [my, Search]
-     * @param numberList e.g. [1, 2]
-     * @return "dc.contributor.author:*my* AND dc.contributor.author:*Search* AND dc.contributor.author:*1* AND ..."
-     */
-    private String composeQueryWithNumbersAndChars(String metadataField, List<String> characterList,
-                                                   List<String> numberList) {
-        this.addQueryTemplateToList(metadataField, characterList);
-        this.addQueryTemplateToList(metadataField, numberList);
-
-        String joinedChars = String.join(" AND ", characterList);
-        String joinedNumbers = String.join(" AND ", numberList);
-        return joinedChars + " AND " + joinedNumbers;
-
-    }
-
-    /**
-     * Add SolrQuery template to the every item of the List
-     * @param metadataField e.g. `dc.contributor.author`
-     * @param stringList could be List of String or List of Numbers which are in the String format because of Solr
-     *                   e.g. [my, Search]
-     * @return [dc.contributor.author:*my*, dc.contributor.author:*Search*]
-     */
-    private List<String> addQueryTemplateToList(String metadataField, List<String> stringList) {
-        String template = metadataField + ":" + "*" + " " + "*";
-
-        AtomicInteger index = new AtomicInteger();
-        stringList.forEach(characters -> {
-            String queryString = template.replaceAll(" ", characters);
-            stringList.set(index.getAndIncrement(), queryString);
-        });
-        return stringList;
-    }
-
-    private DiscoverQuery createDiscoverQuery(String metadataField, String searchValue, Pageable pageable) {
+    private DiscoverQuery createDiscoverQuery(String metadataField, String searchValue) {
         DiscoverQuery discoverQuery = new DiscoverQuery();
         discoverQuery.setQuery(metadataField + ":" + "*" + searchValue + "*");
-        discoverQuery.setStart(Math.toIntExact(pageable.getOffset()));
-        discoverQuery.setMaxResults(pageable.getPageSize());
+        discoverQuery.setMaxResults(500);
```

Comment:
It fetched suggestions only from `pageable.getPageSize()` what is usually about 20 records. 
I hardcoded to fetch suggestion from 500 records, I know this is really not a good solution, but I created and issue to fix it also in the MetadataValueRestRepository, because there is also fetched suggestions in this wrong way. 
Issue: https://github.com/dataquest-dev/DSpace/issues/789

It is inconsistent to fix it here, because this PR is already quite big.